src/components/accounts/SetupImplicit.js

```diff
@@ -134,167 +107,143 @@ class SetupImplicit extends Component {
     }
 
     checkBalance = async () => {
-        const { implicitAccountId } = this.props
-
-        const account = new nearApiJs.Account(this.connection, implicitAccountId)
-
-        await Mixpanel.withTracking("CA Check balance from implicit",
-            async () => {
-                const state = await account.state()
-                if (new BN(state.amount).gte(MIN_BALANCE_TO_CREATE)) {
-                    Mixpanel.track("CA Check balance from implicit: sufficient")
-                    return this.setState({ balance: formatNearAmount(state.amount, 2), whereToBuy: false, createAccount: true })
-                }else {
-                    Mixpanel.track("CA Check balance from implicit: insufficient")
-                }
-            },
-            (e) => { 
-                if (e.message.indexOf('exist while viewing') === -1) {
-                    throw e
-                }
-                this.setState({ balance: false })
+        const { implicitAccountId, dispatch } = this.props
+        const { createAccount } = this.state
+
+        if (!document.hidden) {
```

Comment:
I added a check for `!document.hidden` to make sure the current window tab is active to avoid checking the balance every 2 seconds unless the user is likely to be interacting with the page. 

For example, if the user gets to this page and delays sending funds to the address for a few hours. While at the same time, using a different tab for other things. `document.hidden` will always be `false` as long as the tab is visible/active.